app/services/orderService.js

```diff
@@ -82,9 +80,69 @@ class OrderService {
         strategy.now_buy_times = now_buy_times;
         await strategy.save();
 
-        return { order_id: order_id }
+        return { order_id: inOrder.order_id }
     }
 
+    async sellOrder(strategy_id) {
+        const item = await Strategy.findById(strategy_id);
+        // Access the current price
+        const exchange = new (ccxt)[item.exchange]({
+            apiKey: item.key,
+            secret: item.secret,
+            timeout: 5000,
+        });
+        
+        const tickerRes =  await exchange.fetchTicker(item.symbol);
+        const sell_price = tickerRes.bid;
+
+        // Check if the profit reach the setting take-profit rate
+        const profit = item.quote_total * sell_price - item.base_total;
+        const profit_percentage = profit / item.base_total * 100;
+
+        if(!item.stop_profit_percentage) {
+            return;
+        }
+        // Lower than the setting take-profit rate, exit
+        if (profit_percentage < item.stop_profit_percentage) {
+            return false;
+        }
+        if(!item.drawdown) {
+            return;
+        }
+
+        // When the take-profit level is reached and no peak has been set, sell out
+        if (item.drawdown_status === 'N' || item.drawdown <= 0) {
```

Comment:
Same as the above comments. 

And I can find more rules in the if clause that could be encapsulated.